modules/P60_firmware_bin_extractor.sh

```diff
@@ -246,39 +288,60 @@ binwalking() {
     fi
   fi
 
-  export OUTPUT_DIR_binwalk
-  OUTPUT_DIR_binwalk=$(basename "$FIRMWARE_PATH")
-  OUTPUT_DIR_binwalk="$FIRMWARE_PATH_CP""/""$OUTPUT_DIR_binwalk"_binwalk_emba
+  export OUTPUT_DIR_BINWALK
+  OUTPUT_DIR_BINWALK=$(basename "$FIRMWARE_PATH")
+  OUTPUT_DIR_BINWALK="$FIRMWARE_PATH_CP""/""$OUTPUT_DIR_BINWALK"_binwalk_emba
 
   echo
-  print_output "[*] Extracting firmware to directory $OUTPUT_DIR_binwalk"
+  print_output "[*] Extracting firmware to directory $ORANGE$OUTPUT_DIR_BINWALK$NC"
   # this is not working in background. I have created a new function that gets executed in the background
   # probably there is a more elegant way
-  extract_binwalk_helper &
+  extract_binwalk_helper 1 &
   WAIT_PIDS+=( "$!" )
   wait_for_extractor
   WAIT_PIDS=( )
 
-  # as we probably kill binwalk and to not loose the results we need to execute binwalk in a function 
-  # and read the results from the caller
-  if [[ -f "$TMP_DIR"/binwalker.txt ]] ; then
-    tee -a "$LOG_FILE" < "$TMP_DIR"/binwalker.txt 
-  fi
+  MD5_DONE_DEEP+=( "$(md5sum "$FIRMWARE_PATH" | awk '{print $1}')" )
 }
 
 extract_binwalk_helper() {
+
+  local MATRYOSHKA_="$1"
```

Comment:
It would be better to use `"${1:-}"` here, to be safe. I think it could also be a problem with the strict mode?